contract_payment_mode/models/contract.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    payment_mode_id = fields.Many2one(
+        'payment.mode',
+        string='Payment Mode',
+        domain="[('type', '=', 'sale')]")
+
+    @api.multi
+    def on_change_partner_id(self, partner_id, name):
+        res = super(AccountAnalyticAccount, self).on_change_partner_id(
```

Comment:
In this method self is not called, I only use parner_id parameter, I think is right, but @pedrobaeza can make the point clear.